runtime/JavaScript/src/antlr4/atn/ATNConfigSet.js

```diff
@@ -183,7 +183,7 @@ ATNConfigSet.prototype.hashCode = function() {
 ATNConfigSet.prototype.updateHashCode = function(hash) {
 	if (this.readOnly) {
 		if (this.cachedHashCode === -1) {
-            var hash = new Hash();
+            hash = new Hash();
```

Comment:
mmm... not too sure about this one
I think the intent here is to create a brand new cachedHashCode, not to drop the parameter
suggest to rename the local from hash -> tmpHash

Reply:
Creating a new hash changes the parser output for 3 test cases in `TestSemPredEvalLexer`. Investigation what is correct in this case.

Keeping this thread open.